lib/types.nix

```diff
@@ -302,6 +302,86 @@ rec {
       functor = (defaultFunctor name) // { wrapped = elemType; };
     };
 
+
+    # Adapted version of https://github.com/nix-community/home-manager/blob/249650a07ee2d949fa599f3177a8c234adbd1bee/modules/lib/types-dag.nix
+    dagOf = elemType:
+      let
+        dagContentType = nameType: types.submodule {
+          options = {
+            data = mkOption {
+              type = elemType;
+              description = ''
+                The contents of the DAG entry.
+              '';
+            };
+            after = mkOption {
```

Comment:
Could take the opportunity to change the names of `after` and `before`? E.g., `wantedBy` and `wants` to match the naming in systemd.

Reply:
Systemd also has `after` and `before` actually (see `man systemd.unit`), and those make more sense for a DAG than `wantedBy` and `wants`.